src/test/appdistro/requests.spec.ts

```diff
@@ -0,0 +1,214 @@
+    });
+
+    describe("when request succeeds", () => {
+      it("should return null when response does not contain the token", () => {
+        nock(api.appDistributionOrigin)
+          .get(`/v1alpha/apps/${appId}/jwt`)
+          .reply(200, {});
+        return expect(distribution.getJwtToken()).to.be.eventually.eq(undefined);
+      });
+
+      it("should return the token", () => {
+        const fakeToken = "fake-token";
+        nock(api.appDistributionOrigin)
+          .get(`/v1alpha/apps/${appId}/jwt`)
+          .reply(200, { token: fakeToken });
+        return expect(distribution.getJwtToken()).to.be.eventually.eq(fakeToken);
+      });
+    });
+  });
+
+  describe("pollReleaseIdByHash", () => {
+    describe("when request fails", () => {
+      it("should throw error when retry count >= AppDistributionClient.MAX_POLLING_RETRIES", () => {
+        sandbox.stub(distribution, "getReleaseIdByHash").rejects(new Error("Can't find release"));
+        return expect(
+          distribution.pollReleaseIdByHash("mock-hash", AppDistributionClient.MAX_POLLING_RETRIES)
+        ).to.be.rejectedWith(FirebaseError, "Can't find release");
+      });
+    });
+
+    it("should return release id when request succeeds", () => {
+      const releaseId = "fake-release-id";
+      sandbox.stub(distribution, "getReleaseIdByHash").resolves(releaseId);
+      return expect(
+        distribution.pollReleaseIdByHash("mock-hash", AppDistributionClient.MAX_POLLING_RETRIES)
+      ).to.eventually.eq(releaseId);
+    });
+  });
+
+  describe("getReleaseIdByHash", () => {
+    it("should throw an error when request fails", () => {
+      const fakeHash = "fake-hash";
+      nock(api.appDistributionOrigin)
+        .get(`/v1alpha/apps/${appId}/release_by_hash/${fakeHash}`)
+        .reply(400, {});
+
+      return expect(distribution.getReleaseIdByHash(fakeHash)).to.be.rejectedWith(
+        FirebaseError,
+        "HTTP Error: 400"
+      );
+    });
+
+    describe("when request succeeds", () => {
+      it("should return undefined when it cannot parse the response", () => {
+        const fakeHash = "fake-hash";
+        nock(api.appDistributionOrigin)
+          .get(`/v1alpha/apps/${appId}/release_by_hash/${fakeHash}`)
+          .reply(200, {});
+
+        return expect(distribution.getReleaseIdByHash(fakeHash)).to.eventually.eq(undefined);
+      });
+
+      it("should return the release id", () => {
+        const releaseId = "fake-release-id";
+        const fakeHash = "fake-hash";
+        nock(api.appDistributionOrigin)
+          .get(`/v1alpha/apps/${appId}/release_by_hash/${fakeHash}`)
+          .reply(200, { release: { id: releaseId } });
+
+        return expect(distribution.getReleaseIdByHash(fakeHash)).to.eventually.eq(releaseId);
+      });
+    });
+  });
+
+  describe("addReleaseNotes", () => {
+    it("should return immediately when no release notes are specified", (done) => {
```

Comment:
Rather than using `notify(done)`, prefer using an async function and await:

```typescript
it("should return immediately when no release notes are specified", async () => {
  const apiSpy = sandbox.spy(api, "request");
  await expect(distribution.addReleaseNotes("fake-release-id", "")).to.eventually.be.fulfilled;
  expect(apiSpy).to.be.not.called;
});
```

(applies below as well)